src/useExecuteToolCall.tsx

```diff
@@ -51,81 +184,56 @@ export const useExecuteToolCall = (
             : String(chain.chainID);
       }
 
-      // if operation needs wallet connect
-      // and the current wallet connection isn't one that's included in wantsWallet
-      // we then try to establish that connection
+      // Check if wallet connection is needed and not already established
       if (
         operation.needsWallet &&
         Array.isArray(operation.needsWallet) &&
         !operation.needsWallet.includes(walletStore.getSnapshot().walletType)
       ) {
-        for (const walletType of operation.needsWallet) {
-          await walletStore.connectWallet({
-            walletType,
-            chainId,
-          });
+        setIsWalletConnecting(true);
+        openModal();
 
-          break;
-        }
```

Comment:
We can't do this anymore since we are supporting more than one wallet. We need to await the user's decision in the modal (which wallet they chose) and in the extension (permitting the connection).